lib/datadog/grok/src/parse_grok_rules.rs

```diff
@@ -147,9 +147,14 @@ fn parse_pattern(
 ) -> Result<GrokRule, Error> {
     let parsed_pattern = parse_grok_rule(pattern, aliases, parsed_aliases, &mut Vec::new())?;
     let mut pattern = String::new();
-    pattern.push('^');
+    // (?m) enables DOTALL mode(. includes new lines)
+    // \A, \z - parses from the beginning to the end of string, not line(until \n)
```

Comment:
What is the reason that we change `^` to `\A` and `$` to `\z`? Does `$` only match until a newline (that seems weird, and another oddity of our regex engine)?

The docs over at https://docs.datadoghq.com/logs/log_configuration/parsing/ note:

> The regex matcher applies an implicit `^`, to match the start of a string, and `$`, to match the end of a string.

Reply:
From the docs here https://github.com/kkos/oniguruma/blob/master/doc/RE#L174-L181

In oniguruma, `^` and `$` are the beginning and end of the line.

Reply:
Since we use a different regex engine to Datadog (they use the Java one) we need to account for the various subtle differences. I presume this is one of them.

Reply:
Right, exactly, with ^ and $ instead of \A and \z oniguruma stop parsing when it encounters new line symbols.